### PR TITLE
DBZ-3365 Allow building Oracle connector on CI

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3365

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java [hunk 1]
```diff
@@ -379,9 +379,17 @@ public OracleConnectorConfig(Configuration config) {
         this.logMiningHistoryRecorder = resolveLogMiningHistoryRecorder(config);
         this.jdbcConfig = config.subset(DATABASE_CONFIG_PREFIX, true);
         this.snapshotEnhancementToken = config.getString(SNAPSHOT_ENHANCEMENT_TOKEN);
+        this.connectorAdapter = ConnectorAdapter.parse(config.getString(CONNECTOR_ADAPTER));
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:382]: Could you use the code from `io.debezium.config.Configuration.getInstance(Field, Class<T>, Supplier<ClassLoader>)`?

File: .github/workflows/oracle-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - 'debezium-assembly-descriptors/**'
+      - 'debezium-core/**'
+      - 'debezium-embedded/**'
+      - 'debezium-connector-oracle/**'
+      - 'debezium-microbenchmark/**'
+      - 'debezium-e2e-benchmark/**'
+      - 'debezium-scripting/**'
+      - 'debezium-parent/pom.xml'
+      - 'debezium-bom/pom.xml'
+      - 'pom.xml'
```
Comment (jpechane) [inline .github/workflows/oracle-workflow.yml:20]: Please add the additionl path as in https://github.com/debezium/debezium/pull/2276
Same for the PR paths

**Comment by jpechane** [comment]
@Naros Oracle module is still guraded by oracle profile in the parent POM. Is it still necessary?

**Comment by Naros** [comment]
> Oracle module is still guraded by oracle profile in the parent POM. Is it still necessary?

I think so, although you and @gunnarmorling may think differently.  I left the profile there for now because if a contributor does a `mvn install`, it would then attempt to build & run the Oracle test suite and that would result in a build failure since for the moment the Oracle database has to be built separately in Vagrant.  I think in conjunction with removing the profile, we should also look at investing in spinning up an Oracle 19 docker container like we do for the other connectors from Oracle registry perhaps?

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java (hunk 1 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:382]: I could, but probably not in the same way in which we use this for things such as `database.history` in MySQL.  The property that is being used here is an Enum and so what we'd need to do is actually directly call `Instantiator` directly with the right arguments much like what is done in the `Configuration` interface.  I'll provide a commit to illustrate this and you can let me know if it's acceptable.

Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:383]: See https://github.com/debezium/debezium/pull/2274/commits/d5851862515b6ac8d3bae971183a10bdd7a9e5d7

Comment (jpechane) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java:383]: That's exactly what I had in my mind

**Pull request merged**